jetty-core/jetty-client/src/test/java/org/eclipse/jetty/client/util/MultiPartRequestContentTest.java

```diff
@@ -439,9 +439,9 @@ public boolean handle(Request request, Response response, Callback callback) thr
             String contentType = request.getHeaders().get(HttpHeader.CONTENT_TYPE);
             assertEquals("multipart/form-data", HttpField.valueParameters(contentType, null));
             String boundary = MultiPart.extractBoundary(contentType);
-            MultiPartFormData formData = new MultiPartFormData(boundary);
+            MultiPartFormData formData = new MultiPartFormData(request, boundary);
             formData.setFilesDirectory(tmpDir);
-            formData.parse(request);
+            formData.parse();
```

Comment:
Don't like this because it makes `MultiPartFormData` non-reusable, and it is also weird because a constructor dependency typically implies "I need this component to work", while here the source it's not necessary at all -- it is only necessary when parsing it.

It is also confusing when using `MultiPartFormData` to parse chunks, as it would happen in the client.
What would be the source passed to the constructor used for?